### PR TITLE
[blog] Build a Basic CRUD App with Laravel and Vue

### PR DESCRIPTION
**Pull request by aaronpk**

### CODE REVIEW

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 1]
```diff
@@ -0,0 +1,958 @@
...
+description: "In this tutorial, you'll learn how to build a Vue app using a Laravel API to store its data."
+tags: [crud, vue, php, laravel]
+tweets:
+  - "Are you looking to get started with Laravel and Vue? If so, this tutorial is for you!"
+  - "This tutorial shows you how to build a Vue app with a Laravel backend, complete with authentication!"
+image: blog/featured/okta-vue-tile-books-mouse.jpg
+---
+
+Laravel is one of the most popular web frameworks today because of its elegance, simplicity, and readability. It also boasts one of the largest and most active developer communities. The Laravel community has produced a ton of valuable educational resources, including this one! In this tutorial, you'll build a trivia game as two separate projects: a Laravel API and a Vue frontend (using vue-cli). This approach offers some important benefits:
+
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: Are these meant to be list items below?

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 2]
```diff
@@ -0,0 +1,958 @@
...
+3. Set a descriptive application name, add `http://localhost:8080/implicit/callback` as a **Login redirect URI**, and click **Done**. You can leave the rest of the settings as they are.
+
+4. Finally, copy down the value of the **Client ID** variable. This value will be used in the OpenID Connect flow later on.
+
+## Build Your Laravel and Vue CRUD Application
+
+Now it's time to dig in and build a fun trivia game application! This app will be integrated with a free API for trivia quiz questions and will allow us to set up a list of players, load questions, and mark the players' answers as right or wrong.
+
+Here's what your completed application will look like:
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: Missing Image

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 3]
```diff
@@ -0,0 +1,958 @@
...
+
+You can create your own rules, but here's the general gist of the game:
+
+The game host reads questions to the players and marks their answers
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: List items?

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 4]
```diff
@@ -0,0 +1,958 @@
...
+import VueRouter from 'vue-router'
+
+Vue.config.productionTip = false
+Vue.use(VueRouter)
+
+import Dashboard from './components/Dashboard.vue'
+import Auth from '@okta/okta-vue'
+
+Vue.use(Auth, {
+  issuer: 'https://{YOUR_OKTA_DOMAIN}/oauth2/default',
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: Shoule be:
yourOktaDomain
yourClientId

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 5]
```diff
@@ -0,0 +1,958 @@
...
+        }
+
+        // Attempt authorization with the provided token
+        try {
+
+            // Setup the JWT Verifier
+            $jwtVerifier = (new \Okta\JwtVerifier\JwtVerifierBuilder())
+                            ->setAdaptor(new \Okta\JwtVerifier\Adaptors\SpomkyLabsJose())
+                            ->setAudience('api://default')
+                            ->setClientId('{YOUR_CLIENT_ID}')
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: yourClientId
yourIssuerUrl

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 6]
```diff
@@ -0,0 +1,958 @@
...
+
+You have to enable CORS so you can access your API from the frontend application:
+
+```bash
+composer require barryvdh/laravel-cors
+```
+
+`app/Http/Kernel.php`
+
+```
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: missing php indicator

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md [hunk 7]
```diff
@@ -0,0 +1,958 @@
...
+            this.players[index].isCountUpdating = false
+        }
+    }
+```
+The game is complete now! You now have a basic Laravel API that returns trivia questions to authenticated requests, and a Vue front-end that can log users in and make authenticated requests to the Laravel API.
+
+This is a great start, but there is of course room for improvement. You can improve the code by extracting the common API boilerplate code (retrieving the access token, sending the Authorization header, sending a request and receiving a response) into a service class.
+
+You can find the full code here:
+https://github.com/oktadeveloper/okta-php-laravel-vue-crud-example
```
Comment (leebrandt) [inline _source/_posts/2019-01-15-crud-app-laravel-vue.md]: Should be a link

File: _source/_posts/2019-01-15-crud-app-laravel-vue.md (hunk 1 above)
Comment (aaronpk) [reply _source/_posts/2019-01-15-crud-app-laravel-vue.md]: Yep, thanks google

**Comment by aaronpk** [comment]
Thanks @leebrandt, fixed those!

**Pull request merged**